static/app/utils/reactQuery.tsx

```diff
@@ -0,0 +1,70 @@
+import * as reactQuery from '@tanstack/react-query';
+import {QueryClientConfig} from '@tanstack/react-query';
+
+import RequestError from 'sentry/utils/requestError/requestError';
+import useApi from 'sentry/utils/useApi';
+
+type QueryKeyEndpointOptions = {
+  query?: Record<string, any>;
+};
+
+type QueryKey = readonly [string] | readonly [string, QueryKeyEndpointOptions];
+
+type UseQueryOptions<TQueryFnData, TError, TData, TQueryKey extends QueryKey> = Omit<
+  reactQuery.UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
+  'queryKey' | 'queryFn'
+>;
+
+// We are not overriding any defaults options for stale time, retries, etc.
+// See https://tanstack.com/query/v4/docs/guides/important-defaults
+const DEFAULT_QUERY_CLIENT_CONFIG: QueryClientConfig = {};
```

Comment:
since we are not having any values by default, do we need it now?

Reply:
I think it's good for in the future if we do